src/server.rs

```diff
@@ -407,20 +410,57 @@ impl Server {
                         }),
                     );
 
+                    // Load OS root certificates, if enabled in the configuration.
+                    if config.general.trust_os_certificates {
+                        match get_os_root_certificates() {
+                            Ok(os_certs) => {
+                                let result = root_store.add_parsable_certificates(os_certs);
+
+                                debug!("{} get_os_root_certificates processed {} valid and {} invalid certs", address, result.0, result.1);
+                            }
+                            Err(err) => {
+                                warn!("{} Failed to load OS root certificates: {:?}", address, err)
```

Comment:
This would log a lot. I think we should try to load the store on boot, e.g. during config validation, and only issue a warning once.